pkg/controller/migmigration/restore.go

```diff
@@ -285,22 +309,130 @@ func (t *Task) deleteRestores() error {
 	return nil
 }
 
-// Update namespace mapping for restore
-func (t *Task) updateNamespaceMapping(restore *velero.Restore) {
-	namespaceMapping := make(map[string]string)
-	for _, namespace := range t.namespaces() {
-		mapping := strings.Split(namespace, ":")
-		if len(mapping) == 2 {
-			if mapping[0] == mapping[1] {
-				continue
+func (t *Task) deleteMigrated() error {
+	client, GVRs, err := t.getResourcesForDelete()
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		migapi.MigratedByLabel: string(t.Owner.UID),
+	}).AsListOptions()
+
+	for _, gvr := range GVRs {
+		list, err := client.Resource(gvr).List(*listOptions)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+		for _, r := range list.Items {
+			err = client.Resource(gvr).Namespace(r.GetNamespace()).Delete(r.GetName(), nil)
+			if err != nil {
+				// Will ignore the ones that were removed, or for some reason are not supported
+				// Assuming that main resources will be removed, such as pods and pvcs
+				if k8serror.IsMethodNotSupported(err) || k8serror.IsNotFound(err) {
+					continue
+				}
+				log.Error(err, fmt.Sprintf("Failed to request delete on: %s", gvr.String()))
+				return err
 			}
-			if mapping[1] != "" {
-				namespaceMapping[mapping[0]] = mapping[1]
+		}
+	}
+
+	return nil
+}
+
+func (t *Task) waitForDeleteMigrated() (bool, error) {
+	client, GVRs, err := t.getResourcesForDelete()
+	if err != nil {
+		log.Trace(err)
+		return false, err
+	}
+
+	listOptions := k8sclient.MatchingLabels(map[string]string{
+		migapi.MigratedByLabel: string(t.Owner.UID),
+	}).AsListOptions()
+	for _, gvr := range GVRs {
+		list, err := client.Resource(gvr).List(*listOptions)
+		if err != nil {
+			log.Trace(err)
+			return false, err
+		}
+		// Wait for resources with deletion timestamps
+		for _, r := range list.Items {
+			if r.GetDeletionTimestamp() != nil {
+				return false, err
 			}
 		}
 	}
 
-	if len(namespaceMapping) != 0 {
-		restore.Spec.NamespaceMapping = namespaceMapping
+	return true, nil
+}
+
+func (t *Task) deleteLabels() error {
```

Comment:
All of the other annotation & label management methods are in `annotation.go`.
Please follow the established convention of organizing functionality in `go` files.

Reply:
Sounds good, fixing.